PR for new model eos6758

arnaucoma24 opened the pull request.

GemmaTuron commented:
@arnaucoma24 

Please double check the metadata.yml file

arnaucoma24 commented:
Thanks @GemmaTuron - metadata.yml is now updated

arnaucoma24 commented:
Are the workflows expecting a "card" key in metadata.yml?

GemmaTuron commented:
@arnaucoma24 

First, have you tried fetching the model from run.sh first and from_dir next to ensure it works before opening the PR?
The workflows are not looking for a card field in the metadata, the card contains all model information, this is the exact error:
```
  File "/usr/share/miniconda/lib/python3.12/site-packages/ersilia/io/input.py", line 76, in _read_input_from_card
    input_type = self.mc.get(model_id)["card"]["Input"]
                 ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
```

arnaucoma24 commented:
Hi @GemmaTuron 

Indeed, I run the run.sh file to create the run_output.csv and everything was smooth. However, I did not try to fetch the model from_dir. I am doing it now, and just came across what seems to be a problem related with the previous error (see the logs below). If I understood it correctly, ersilia pack is looking for 'Input Shape' when converting metadata.yml to json format - but we now have "Input Dimension" instead. This is weird, since 'Input Shape' is included in the metadata.json from [eos5axz](https://github.com/ersilia-os/eos5axz/blob/main/metadata.json) (our reference) but not in the metadata.yml from the [eos-template repo.](https://github.com/ersilia-os/eos-template/blob/main/metadata.yml). Do you thing it's somehow related? :crossed_fingers: 

```
(ersilia) acomajuncosa@herbert:~/Documents$ ersilia -v fetch eos6758 --from_dir ./eos6758
15:58:52 | DEBUG    | Repo path specified: ./eos6758
15:58:52 | DEBUG    | Absolute path: /home/acomajuncosa/Documents/eos6758
15:58:52 | DEBUG    | Path exists: /home/acomajuncosa/Documents/eos6758
15:58:52 | DEBUG    | Model ID from path: eos6758
15:58:52 | DEBUG    | Slug from path: mole-representations
⬇️  Fetching model eos6758: mole-representations
15:58:52 | DEBUG    | Initialized with URL: None
15:58:52 | DEBUG    | Getting model source
15:58:52 | DEBUG    | Model getting fetched from Local Repository
15:58:52 | INFO     | Model doesn't exist on your system, fetching it now.
15:58:52 | DEBUG    | Starting fetching procedure
15:58:52 | DEBUG    | Fetching in your system, not from DockerHub
15:58:52 | DEBUG    | Deciding fetcher (BentoML or FastAPI)
15:58:52 | DEBUG    | Fetching using Ersilia Pack (FastAPI)
15:58:52 | DEBUG    | Fetching from FastAPI...
15:58:52 | INFO     | GitHub CLI is not installed. Ersilia can work without it, but we highly recommend that you install this tool.
15:58:52 | DEBUG    | Checking Git LFS
15:58:52 | DEBUG    | Git LFS is installed
Git LFS initialized.
15:58:52 | DEBUG    | Git LFS has been activated
15:58:52 | DEBUG    | Connected to the internet
15:58:52 | DEBUG    | Conda is installed
15:58:52 | DEBUG    | EOS Home path exists
15:58:52 | INFO     | Starting delete of model eos6758
15:58:52 | INFO     | Removing EOS folder /home/acomajuncosa/eos/dest/eos6758
15:58:52 | DEBUG    | Attempting Bento delete
15:58:52 | DEBUG    | BentoML is not installed
15:58:52 | DEBUG    | BentoML is not installed, no need for deleting using BentoML
15:58:52 | INFO     | Deleting conda environment eos6758
Remove all packages in environment /home/acomajuncosa/miniconda3/envs/eos6758:


## Package Plan ##

  environment location: /home/acomajuncosa/miniconda3/envs/eos6758


The following packages will be REMOVED:

  _libgcc_mutex-0.1-main
  _openmp_mutex-5.1-1_gnu
  bzip2-1.0.8-h5eee18b_6
  ca-certificates-2025.2.25-h06a4308_0
  ld_impl_linux-64-2.40-h12ee557_0
  libffi-3.4.4-h6a678d5_1
  libgcc-ng-11.2.0-h1234567_1
  libgomp-11.2.0-h1234567_1
  libstdcxx-ng-11.2.0-h1234567_1
  libuuid-1.41.5-h5eee18b_0
  ncurses-6.4-h6a678d5_0
  openssl-3.0.16-h5eee18b_0
  pip-25.0-py310h06a4308_0
  python-3.10.16-he870216_1
  readline-8.2-h5eee18b_0
  setuptools-75.8.0-py310h06a4308_0
  sqlite-3.45.3-h5eee18b_0
  tk-8.6.14-h39e8969_0
  tzdata-2025a-h04d1e81_0
  wheel-0.45.1-py310h06a4308_0
  xz-5.6.4-h5eee18b_1
  zlib-1.2.13-h5eee18b_1



Downloading and Extracting Packages: ...working... done
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
15:58:56 | DEBUG    | Attempting temporary folder delete
15:58:56 | INFO     | Removing docker images and stopping containers related to eos6758
15:58:56 | DEBUG    | Model entry eos6758 was not available in the fetched models registry
15:58:56 | SUCCESS  | Model eos6758 deleted successfully
15:58:56 | DEBUG    | Getting repository
15:58:57 | DEBUG    | User is not root
15:58:57 | DEBUG    | Copying example file if available
15:58:57 | DEBUG    | Example file exists
15:58:57 | DEBUG    | Done getting model
15:58:57 | DEBUG    | Check if model can be run with vanilla (system) code. This is a default option when inside a docker container.
15:58:57 | DEBUG    | Initializing conda packer
15:58:57 | DEBUG    | Packing model with Conda
15:58:57 | DEBUG    | Setting up
15:58:57 | DEBUG    | Reading python version from install.yml
15:58:57 | DEBUG    | Conda environment eos6758
15:58:57 | DEBUG    | Environment eos6758 does not exist
Retrieving notices: done
Channels:
 - defaults
Platform: linux-64
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done

## Package Plan ##

  environment location: /home/acomajuncosa/miniconda3/envs/eos6758

  added / updated specs:
    - python=3.10


The following NEW packages will be INSTALLED:

  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main 
  _openmp_mutex      pkgs/main/linux-64::_openmp_mutex-5.1-1_gnu 
  bzip2              pkgs/main/linux-64::bzip2-1.0.8-h5eee18b_6 
  ca-certificates    pkgs/main/linux-64::ca-certificates-2025.2.25-h06a4308_0 
  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.40-h12ee557_0 
  libffi             pkgs/main/linux-64::libffi-3.4.4-h6a678d5_1 
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-11.2.0-h1234567_1 
  libgomp            pkgs/main/linux-64::libgomp-11.2.0-h1234567_1 
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-11.2.0-h1234567_1 
  libuuid            pkgs/main/linux-64::libuuid-1.41.5-h5eee18b_0 
  ncurses            pkgs/main/linux-64::ncurses-6.4-h6a678d5_0 
  openssl            pkgs/main/linux-64::openssl-3.0.16-h5eee18b_0 
  pip                pkgs/main/linux-64::pip-25.0-py310h06a4308_0 
  python             pkgs/main/linux-64::python-3.10.16-he870216_1 
  readline           pkgs/main/linux-64::readline-8.2-h5eee18b_0 
  setuptools         pkgs/main/linux-64::setuptools-75.8.0-py310h06a4308_0 
  sqlite             pkgs/main/linux-64::sqlite-3.45.3-h5eee18b_0 
  tk                 pkgs/main/linux-64::tk-8.6.14-h39e8969_0 
  tzdata             pkgs/main/noarch::tzdata-2025a-h04d1e81_0 
  wheel              pkgs/main/linux-64::wheel-0.45.1-py310h06a4308_0 
  xz                 pkgs/main/linux-64::xz-5.6.4-h5eee18b_1 
  zlib               pkgs/main/linux-64::zlib-1.2.13-h5eee18b_1 



Downloading and Extracting Packages: ...working... done
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate eos6758
#
# To deactivate an active environment, use
#
#     $ conda deactivate
==> WARNING: A newer version of conda exists. <==
    current version: 25.1.1
    latest version: 25.3.1

Please update conda by running

    $ conda update -n base -c defaults conda
15:59:01 | DEBUG    | Creating base conda environment
15:59:01 | DEBUG    | Run commandlines on eos6758
15:59:01 | DEBUG    | python -m pip install git+https://github.com/ersilia-os/ersilia-pack.git@fix/removing-prometheus
15:59:02 | DEBUG    | Activating base environment
15:59:02 | DEBUG    | Current working directory: /home/acomajuncosa/Documents
15:59:02 | DEBUG    | Running bash /tmp/ersilia-yjdk8rtd/script.sh 2>&1 | tee -a /tmp/ersilia-tsu53vr1/command_outputs.log
Collecting git+https://github.com/ersilia-os/ersilia-pack.git@fix/removing-prometheus
  Cloning https://github.com/ersilia-os/ersilia-pack.git (to revision fix/removing-prometheus) to /tmp/pip-req-build-djpty3mw
  Running command git clone --filter=blob:none --quiet https://github.com/ersilia-os/ersilia-pack.git /tmp/pip-req-build-djpty3mw
  Running command git checkout -b fix/removing-prometheus --track origin/fix/removing-prometheus
  Switched to a new branch 'fix/removing-prometheus'
  branch 'fix/removing-prometheus' set up to track 'origin/fix/removing-prometheus'.
  Resolved https://github.com/ersilia-os/ersilia-pack.git to commit 47cb900f79218b3938c5922d0ede488593c9a6f1
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting fastapi (from ersilia-pack==1.0.0)
  Using cached fastapi-0.115.12-py3-none-any.whl.metadata (27 kB)
Collecting PyYAML (from ersilia-pack==1.0.0)
  Using cached PyYAML-6.0.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.1 kB)
Collecting uvicorn[standard] (from ersilia-pack==1.0.0)
  Downloading uvicorn-0.34.2-py3-none-any.whl.metadata (6.5 kB)
Collecting slowapi (from ersilia-pack==1.0.0)
  Using cached slowapi-0.1.9-py3-none-any.whl.metadata (3.0 kB)
Collecting pybreaker (from ersilia-pack==1.0.0)
  Using cached pybreaker-1.3.0-py3-none-any.whl.metadata (10 kB)
Collecting redis (from ersilia-pack==1.0.0)
  Using cached redis-5.2.1-py3-none-any.whl.metadata (9.1 kB)
Collecting psutil (from ersilia-pack==1.0.0)
  Using cached psutil-7.0.0-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (22 kB)
Collecting orjson (from ersilia-pack==1.0.0)
  Using cached orjson-3.10.16-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (41 kB)
Collecting starlette<0.47.0,>=0.40.0 (from fastapi->ersilia-pack==1.0.0)
  Downloading starlette-0.46.2-py3-none-any.whl.metadata (6.2 kB)
Collecting pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4 (from fastapi->ersilia-pack==1.0.0)
  Using cached pydantic-2.11.3-py3-none-any.whl.metadata (65 kB)
Collecting typing-extensions>=4.8.0 (from fastapi->ersilia-pack==1.0.0)
  Using cached typing_extensions-4.13.2-py3-none-any.whl.metadata (3.0 kB)
Collecting async-timeout>=4.0.3 (from redis->ersilia-pack==1.0.0)
  Using cached async_timeout-5.0.1-py3-none-any.whl.metadata (5.1 kB)
Collecting limits>=2.3 (from slowapi->ersilia-pack==1.0.0)
  Downloading limits-5.0.0-py3-none-any.whl.metadata (10 kB)
Collecting click>=7.0 (from uvicorn[standard]->ersilia-pack==1.0.0)
  Using cached click-8.1.8-py3-none-any.whl.metadata (2.3 kB)
Collecting h11>=0.8 (from uvicorn[standard]->ersilia-pack==1.0.0)
  Using cached h11-0.14.0-py3-none-any.whl.metadata (8.2 kB)
Collecting httptools>=0.6.3 (from uvicorn[standard]->ersilia-pack==1.0.0)
  Using cached httptools-0.6.4-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.6 kB)
Collecting python-dotenv>=0.13 (from uvicorn[standard]->ersilia-pack==1.0.0)
  Using cached python_dotenv-1.1.0-py3-none-any.whl.metadata (24 kB)
Collecting uvloop!=0.15.0,!=0.15.1,>=0.14.0 (from uvicorn[standard]->ersilia-pack==1.0.0)
  Using cached uvloop-0.21.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.9 kB)
Collecting watchfiles>=0.13 (from uvicorn[standard]->ersilia-pack==1.0.0)
  Using cached watchfiles-1.0.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.9 kB)
Collecting websockets>=10.4 (from uvicorn[standard]->ersilia-pack==1.0.0)
  Using cached websockets-15.0.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.8 kB)
Collecting deprecated>=1.2 (from limits>=2.3->slowapi->ersilia-pack==1.0.0)
  Using cached Deprecated-1.2.18-py2.py3-none-any.whl.metadata (5.7 kB)
Collecting packaging<25,>=21 (from limits>=2.3->slowapi->ersilia-pack==1.0.0)
  Using cached packaging-24.2-py3-none-any.whl.metadata (3.2 kB)
Collecting annotated-types>=0.6.0 (from pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4->fastapi->ersilia-pack==1.0.0)
  Using cached annotated_types-0.7.0-py3-none-any.whl.metadata (15 kB)
Collecting pydantic-core==2.33.1 (from pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4->fastapi->ersilia-pack==1.0.0)
  Using cached pydantic_core-2.33.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.8 kB)
Collecting typing-inspection>=0.4.0 (from pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4->fastapi->ersilia-pack==1.0.0)
  Using cached typing_inspection-0.4.0-py3-none-any.whl.metadata (2.6 kB)
Collecting anyio<5,>=3.6.2 (from starlette<0.47.0,>=0.40.0->fastapi->ersilia-pack==1.0.0)
  Using cached anyio-4.9.0-py3-none-any.whl.metadata (4.7 kB)
Collecting exceptiongroup>=1.0.2 (from anyio<5,>=3.6.2->starlette<0.47.0,>=0.40.0->fastapi->ersilia-pack==1.0.0)
  Using cached exceptiongroup-1.2.2-py3-none-any.whl.metadata (6.6 kB)
Collecting idna>=2.8 (from anyio<5,>=3.6.2->starlette<0.47.0,>=0.40.0->fastapi->ersilia-pack==1.0.0)
  Using cached idna-3.10-py3-none-any.whl.metadata (10 kB)
Collecting sniffio>=1.1 (from anyio<5,>=3.6.2->starlette<0.47.0,>=0.40.0->fastapi->ersilia-pack==1.0.0)
  Using cached sniffio-1.3.1-py3-none-any.whl.metadata (3.9 kB)
Collecting wrapt<2,>=1.10 (from deprecated>=1.2->limits>=2.3->slowapi->ersilia-pack==1.0.0)
  Using cached wrapt-1.17.2-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.4 kB)
Using cached fastapi-0.115.12-py3-none-any.whl (95 kB)
Using cached orjson-3.10.16-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (132 kB)
Using cached psutil-7.0.0-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (277 kB)
Using cached pybreaker-1.3.0-py3-none-any.whl (12 kB)
Using cached PyYAML-6.0.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (751 kB)
Using cached redis-5.2.1-py3-none-any.whl (261 kB)
Using cached slowapi-0.1.9-py3-none-any.whl (14 kB)
Using cached async_timeout-5.0.1-py3-none-any.whl (6.2 kB)
Using cached click-8.1.8-py3-none-any.whl (98 kB)
Using cached h11-0.14.0-py3-none-any.whl (58 kB)
Using cached httptools-0.6.4-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (442 kB)
Downloading limits-5.0.0-py3-none-any.whl (60 kB)
Using cached pydantic-2.11.3-py3-none-any.whl (443 kB)
Using cached pydantic_core-2.33.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.0 MB)
Using cached python_dotenv-1.1.0-py3-none-any.whl (20 kB)
Downloading starlette-0.46.2-py3-none-any.whl (72 kB)
Using cached typing_extensions-4.13.2-py3-none-any.whl (45 kB)
Using cached uvloop-0.21.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.8 MB)
Using cached watchfiles-1.0.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (454 kB)
Using cached websockets-15.0.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (181 kB)
Downloading uvicorn-0.34.2-py3-none-any.whl (62 kB)
Using cached annotated_types-0.7.0-py3-none-any.whl (13 kB)
Using cached anyio-4.9.0-py3-none-any.whl (100 kB)
Using cached Deprecated-1.2.18-py2.py3-none-any.whl (10.0 kB)
Using cached packaging-24.2-py3-none-any.whl (65 kB)
Using cached typing_inspection-0.4.0-py3-none-any.whl (14 kB)
Using cached exceptiongroup-1.2.2-py3-none-any.whl (16 kB)
Using cached idna-3.10-py3-none-any.whl (70 kB)
Using cached sniffio-1.3.1-py3-none-any.whl (10 kB)
Using cached wrapt-1.17.2-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (82 kB)
Building wheels for collected packages: ersilia-pack
  Building wheel for ersilia-pack (pyproject.toml): started
  Building wheel for ersilia-pack (pyproject.toml): finished with status 'done'
  Created wheel for ersilia-pack: filename=ersilia_pack-1.0.0-py3-none-any.whl size=42361 sha256=e2d6c2cbb58d623a954c5ac38492de3fc1868b99faa4dfc84576106c34ce6805
  Stored in directory: /tmp/pip-ephem-wheel-cache-aw8u0arw/wheels/45/2f/3f/3c7ab258fda297bffcab3751c95a080732dcedaebfff37bcf9
Successfully built ersilia-pack
Installing collected packages: wrapt, websockets, uvloop, typing-extensions, sniffio, PyYAML, python-dotenv, pybreaker, psutil, packaging, orjson, idna, httptools, h11, exceptiongroup, click, async-timeout, annotated-types, uvicorn, typing-inspection, redis, pydantic-core, deprecated, anyio, watchfiles, starlette, pydantic, limits, slowapi, fastapi, ersilia-pack
Successfully installed PyYAML-6.0.2 annotated-types-0.7.0 anyio-4.9.0 async-timeout-5.0.1 click-8.1.8 deprecated-1.2.18 ersilia-pack-1.0.0 exceptiongroup-1.2.2 fastapi-0.115.12 h11-0.14.0 httptools-0.6.4 idna-3.10 limits-5.0.0 orjson-3.10.16 packaging-24.2 psutil-7.0.0 pybreaker-1.3.0 pydantic-2.11.3 pydantic-core-2.33.1 python-dotenv-1.1.0 redis-5.2.1 slowapi-0.1.9 sniffio-1.3.1 starlette-0.46.2 typing-extensions-4.13.2 typing-inspection-0.4.0 uvicorn-0.34.2 uvloop-0.21.0 watchfiles-1.0.5 websockets-15.0.1 wrapt-1.17.2
15:59:11 | DEBUG    | Collecting git+https://github.com/ersilia-os/ersilia-pack.git@fix/removing-prometheus
  Cloning https://github.com/ersilia-os/ersilia-pack.git (to revision fix/removing-prometheus) to /tmp/pip-req-build-djpty3mw
  Running command git clone --filter=blob:none --quiet https://github.com/ersilia-os/ersilia-pack.git /tmp/pip-req-build-djpty3mw
  Running command git checkout -b fix/removing-prometheus --track origin/fix/removing-prometheus
  Switched to a new branch 'fix/removing-prometheus'
  branch 'fix/removing-prometheus' set up to track 'origin/fix/removing-prometheus'.
  Resolved https://github.com/ersilia-os/ersilia-pack.git to commit 47cb900f79218b3938c5922d0ede488593c9a6f1
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting fastapi (from ersilia-pack==1.0.0)
  Using cached fastapi-0.115.12-py3-none-any.whl.metadata (27 kB)
Collecting PyYAML (from ersilia-pack==1.0.0)
  Using cached PyYAML-6.0.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.1 kB)
Collecting uvicorn[standard] (from ersilia-pack==1.0.0)
  Downloading uvicorn-0.34.2-py3-none-any.whl.metadata (6.5 kB)
Collecting slowapi (from ersilia-pack==1.0.0)
  Using cached slowapi-0.1.9-py3-none-any.whl.metadata (3.0 kB)
Collecting pybreaker (from ersilia-pack==1.0.0)
  Using cached pybreaker-1.3.0-py3-none-any.whl.metadata (10 kB)
Collecting redis (from ersilia-pack==1.0.0)
  Using cached redis-5.2.1-py3-none-any.whl.metadata (9.1 kB)
Collecting psutil (from ersilia-pack==1.0.0)
  Using cached psutil-7.0.0-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (22 kB)
Collecting orjson (from ersilia-pack==1.0.0)
  Using cached orjson-3.10.16-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (41 kB)
Collecting starlette<0.47.0,>=0.40.0 (from fastapi->ersilia-pack==1.0.0)
  Downloading starlette-0.46.2-py3-none-any.whl.metadata (6.2 kB)
Collecting pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4 (from fastapi->ersilia-pack==1.0.0)
  Using cached pydantic-2.11.3-py3-none-any.whl.metadata (65 kB)
Collecting typing-extensions>=4.8.0 (from fastapi->ersilia-pack==1.0.0)
  Using cached typing_extensions-4.13.2-py3-none-any.whl.metadata (3.0 kB)
Collecting async-timeout>=4.0.3 (from redis->ersilia-pack==1.0.0)
  Using cached async_timeout-5.0.1-py3-none-any.whl.metadata (5.1 kB)
Collecting limits>=2.3 (from slowapi->ersilia-pack==1.0.0)
  Downloading limits-5.0.0-py3-none-any.whl.metadata (10 kB)
Collecting click>=7.0 (from uvicorn[standard]->ersilia-pack==1.0.0)
  Using cached click-8.1.8-py3-none-any.whl.metadata (2.3 kB)
Collecting h11>=0.8 (from uvicorn[standard]->ersilia-pack==1.0.0)
  Using cached h11-0.14.0-py3-none-any.whl.metadata (8.2 kB)
Collecting httptools>=0.6.3 (from uvicorn[standard]->ersilia-pack==1.0.0)
  Using cached httptools-0.6.4-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.6 kB)
Collecting python-dotenv>=0.13 (from uvicorn[standard]->ersilia-pack==1.0.0)
  Using cached python_dotenv-1.1.0-py3-none-any.whl.metadata (24 kB)
Collecting uvloop!=0.15.0,!=0.15.1,>=0.14.0 (from uvicorn[standard]->ersilia-pack==1.0.0)
  Using cached uvloop-0.21.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.9 kB)
Collecting watchfiles>=0.13 (from uvicorn[standard]->ersilia-pack==1.0.0)
  Using cached watchfiles-1.0.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.9 kB)
Collecting websockets>=10.4 (from uvicorn[standard]->ersilia-pack==1.0.0)
  Using cached websockets-15.0.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.8 kB)
Collecting deprecated>=1.2 (from limits>=2.3->slowapi->ersilia-pack==1.0.0)
  Using cached Deprecated-1.2.18-py2.py3-none-any.whl.metadata (5.7 kB)
Collecting packaging<25,>=21 (from limits>=2.3->slowapi->ersilia-pack==1.0.0)
  Using cached packaging-24.2-py3-none-any.whl.metadata (3.2 kB)
Collecting annotated-types>=0.6.0 (from pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4->fastapi->ersilia-pack==1.0.0)
  Using cached annotated_types-0.7.0-py3-none-any.whl.metadata (15 kB)
Collecting pydantic-core==2.33.1 (from pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4->fastapi->ersilia-pack==1.0.0)
  Using cached pydantic_core-2.33.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.8 kB)
Collecting typing-inspection>=0.4.0 (from pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4->fastapi->ersilia-pack==1.0.0)
  Using cached typing_inspection-0.4.0-py3-none-any.whl.metadata (2.6 kB)
Collecting anyio<5,>=3.6.2 (from starlette<0.47.0,>=0.40.0->fastapi->ersilia-pack==1.0.0)
  Using cached anyio-4.9.0-py3-none-any.whl.metadata (4.7 kB)
Collecting exceptiongroup>=1.0.2 (from anyio<5,>=3.6.2->starlette<0.47.0,>=0.40.0->fastapi->ersilia-pack==1.0.0)
  Using cached exceptiongroup-1.2.2-py3-none-any.whl.metadata (6.6 kB)
Collecting idna>=2.8 (from anyio<5,>=3.6.2->starlette<0.47.0,>=0.40.0->fastapi->ersilia-pack==1.0.0)
  Using cached idna-3.10-py3-none-any.whl.metadata (10 kB)
Collecting sniffio>=1.1 (from anyio<5,>=3.6.2->starlette<0.47.0,>=0.40.0->fastapi->ersilia-pack==1.0.0)
  Using cached sniffio-1.3.1-py3-none-any.whl.metadata (3.9 kB)
Collecting wrapt<2,>=1.10 (from deprecated>=1.2->limits>=2.3->slowapi->ersilia-pack==1.0.0)
  Using cached wrapt-1.17.2-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.4 kB)
Using cached fastapi-0.115.12-py3-none-any.whl (95 kB)
Using cached orjson-3.10.16-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (132 kB)
Using cached psutil-7.0.0-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (277 kB)
Using cached pybreaker-1.3.0-py3-none-any.whl (12 kB)
Using cached PyYAML-6.0.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (751 kB)
Using cached redis-5.2.1-py3-none-any.whl (261 kB)
Using cached slowapi-0.1.9-py3-none-any.whl (14 kB)
Using cached async_timeout-5.0.1-py3-none-any.whl (6.2 kB)
Using cached click-8.1.8-py3-none-any.whl (98 kB)
Using cached h11-0.14.0-py3-none-any.whl (58 kB)
Using cached httptools-0.6.4-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (442 kB)
Downloading limits-5.0.0-py3-none-any.whl (60 kB)
Using cached pydantic-2.11.3-py3-none-any.whl (443 kB)
Using cached pydantic_core-2.33.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.0 MB)
Using cached python_dotenv-1.1.0-py3-none-any.whl (20 kB)
Downloading starlette-0.46.2-py3-none-any.whl (72 kB)
Using cached typing_extensions-4.13.2-py3-none-any.whl (45 kB)
Using cached uvloop-0.21.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.8 MB)
Using cached watchfiles-1.0.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (454 kB)
Using cached websockets-15.0.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (181 kB)
Downloading uvicorn-0.34.2-py3-none-any.whl (62 kB)
Using cached annotated_types-0.7.0-py3-none-any.whl (13 kB)
Using cached anyio-4.9.0-py3-none-any.whl (100 kB)
Using cached Deprecated-1.2.18-py2.py3-none-any.whl (10.0 kB)
Using cached packaging-24.2-py3-none-any.whl (65 kB)
Using cached typing_inspection-0.4.0-py3-none-any.whl (14 kB)
Using cached exceptiongroup-1.2.2-py3-none-any.whl (16 kB)
Using cached idna-3.10-py3-none-any.whl (70 kB)
Using cached sniffio-1.3.1-py3-none-any.whl (10 kB)
Using cached wrapt-1.17.2-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (82 kB)
Building wheels for collected packages: ersilia-pack
  Building wheel for ersilia-pack (pyproject.toml): started
  Building wheel for ersilia-pack (pyproject.toml): finished with status 'done'
  Created wheel for ersilia-pack: filename=ersilia_pack-1.0.0-py3-none-any.whl size=42361 sha256=e2d6c2cbb58d623a954c5ac38492de3fc1868b99faa4dfc84576106c34ce6805
  Stored in directory: /tmp/pip-ephem-wheel-cache-aw8u0arw/wheels/45/2f/3f/3c7ab258fda297bffcab3751c95a080732dcedaebfff37bcf9
Successfully built ersilia-pack
Installing collected packages: wrapt, websockets, uvloop, typing-extensions, sniffio, PyYAML, python-dotenv, pybreaker, psutil, packaging, orjson, idna, httptools, h11, exceptiongroup, click, async-timeout, annotated-types, uvicorn, typing-inspection, redis, pydantic-core, deprecated, anyio, watchfiles, starlette, pydantic, limits, slowapi, fastapi, ersilia-pack
Successfully installed PyYAML-6.0.2 annotated-types-0.7.0 anyio-4.9.0 async-timeout-5.0.1 click-8.1.8 deprecated-1.2.18 ersilia-pack-1.0.0 exceptiongroup-1.2.2 fastapi-0.115.12 h11-0.14.0 httptools-0.6.4 idna-3.10 limits-5.0.0 orjson-3.10.16 packaging-24.2 psutil-7.0.0 pybreaker-1.3.0 pydantic-2.11.3 pydantic-core-2.33.1 python-dotenv-1.1.0 redis-5.2.1 slowapi-0.1.9 sniffio-1.3.1 starlette-0.46.2 typing-extensions-4.13.2 typing-inspection-0.4.0 uvicorn-0.34.2 uvloop-0.21.0 watchfiles-1.0.5 websockets-15.0.1 wrapt-1.17.2

15:59:11 | DEBUG    | Activation done
15:59:11 | DEBUG    | Storing Conda environment in the local environment database
15:59:11 | DEBUG    | Done with the Conda setup
15:59:12 | DEBUG    | Using environment eos6758
15:59:12 | DEBUG    | Running command: ersilia_model_pack --repo_path /home/acomajuncosa/eos/dest/eos6758 --bundles_repo_path /home/acomajuncosa/eos/repository --conda_env_name eos6758
15:59:12 | DEBUG    | Run commandlines on eos6758
15:59:12 | DEBUG    | ersilia_model_pack --repo_path /home/acomajuncosa/eos/dest/eos6758 --bundles_repo_path /home/acomajuncosa/eos/repository --conda_env_name eos6758
15:59:13 | DEBUG    | Activating base environment
15:59:13 | DEBUG    | Current working directory: /home/acomajuncosa/Documents
15:59:13 | DEBUG    | Running bash /tmp/ersilia-tp2das0e/script.sh 2>&1 | tee -a /tmp/ersilia-4911b1i4/command_outputs.log
Traceback (most recent call last):
  File "/home/acomajuncosa/miniconda3/envs/eos6758/bin/ersilia_model_pack", line 8, in <module>
    sys.exit(main())
  File "/home/acomajuncosa/miniconda3/envs/eos6758/lib/python3.10/site-packages/ersilia_pack/packer.py", line 366, in main
    fp = FastApiAppPacker(args.repo_path, args.bundles_repo_path, args.conda_env_name)
  File "/home/acomajuncosa/miniconda3/envs/eos6758/lib/python3.10/site-packages/ersilia_pack/packer.py", line 31, in __init__
    self.model_id = self._get_model_id()
  File "/home/acomajuncosa/miniconda3/envs/eos6758/lib/python3.10/site-packages/ersilia_pack/packer.py", line 53, in _get_model_id
    data = self._load_metadata()
  File "/home/acomajuncosa/miniconda3/envs/eos6758/lib/python3.10/site-packages/ersilia_pack/packer.py", line 93, in _load_metadata
    data = self._load_metadata_as_dict_from_file()
  File "/home/acomajuncosa/miniconda3/envs/eos6758/lib/python3.10/site-packages/ersilia_pack/packer.py", line 88, in _load_metadata_as_dict_from_file
    data = MetadataYml2JsonConverter(yml_file).convert()
  File "/home/acomajuncosa/miniconda3/envs/eos6758/lib/python3.10/site-packages/ersilia_pack/parsers/metadata_yml2json_converter.py", line 34, in convert
    data["Input Shape"] = self._tostr(self.data["Input Shape"])
KeyError: 'Input Shape'
15:59:13 | DEBUG    | Traceback (most recent call last):
  File "/home/acomajuncosa/miniconda3/envs/eos6758/bin/ersilia_model_pack", line 8, in <module>
    sys.exit(main())
  File "/home/acomajuncosa/miniconda3/envs/eos6758/lib/python3.10/site-packages/ersilia_pack/packer.py", line 366, in main
    fp = FastApiAppPacker(args.repo_path, args.bundles_repo_path, args.conda_env_name)
  File "/home/acomajuncosa/miniconda3/envs/eos6758/lib/python3.10/site-packages/ersilia_pack/packer.py", line 31, in __init__
    self.model_id = self._get_model_id()
  File "/home/acomajuncosa/miniconda3/envs/eos6758/lib/python3.10/site-packages/ersilia_pack/packer.py", line 53, in _get_model_id
    data = self._load_metadata()
  File "/home/acomajuncosa/miniconda3/envs/eos6758/lib/python3.10/site-packages/ersilia_pack/packer.py", line 93, in _load_metadata
    data = self._load_metadata_as_dict_from_file()
  File "/home/acomajuncosa/miniconda3/envs/eos6758/lib/python3.10/site-packages/ersilia_pack/packer.py", line 88, in _load_metadata_as_dict_from_file
    data = MetadataYml2JsonConverter(yml_file).convert()
  File "/home/acomajuncosa/miniconda3/envs/eos6758/lib/python3.10/site-packages/ersilia_pack/parsers/metadata_yml2json_converter.py", line 34, in convert
    data["Input Shape"] = self._tostr(self.data["Input Shape"])
KeyError: 'Input Shape'

15:59:13 | DEBUG    | Activation done
15:59:13 | DEBUG    | Packing command successfully run inside eos6758 conda environment
15:59:13 | DEBUG    | Creating model symlink bundle > dest
15:59:14 | DEBUG    | Bundle path not found: /home/acomajuncosa/eos/repository/eos6758
🚨🚨🚨 Something went wrong with Ersilia 🚨🚨🚨

Error message:

expected str, bytes or os.PathLike object, not NoneType
If this error message is not helpful, open an issue at:
 - https://github.com/ersilia-os/ersilia
Or feel free to reach out to us at:
 - hello[at]ersilia.io

If you haven't, try to run your command in verbose mode (-v in the CLI)
 - You will find the console log file in: /home/acomajuncosa/eos/current.log
Task exception was never retrieved
future: <Task finished name='Task-1' coro=<ModelFetcher.fetch() done, defined at /home/acomajuncosa/Documents/ersilia/ersilia/hub/fetch/fetch.py:344> exception=SystemExit(1)>
Traceback (most recent call last):
  File "/home/acomajuncosa/Documents/ersilia/ersilia/utils/exceptions_utils/throw_ersilia_exception.py", line 25, in inner_function
    return func(*args, **kwargs)
  File "/home/acomajuncosa/Documents/ersilia/ersilia/hub/fetch/pack/fastapi_pack/runners.py", line 156, in run
    self._run()
  File "/home/acomajuncosa/Documents/ersilia/ersilia/hub/fetch/pack/fastapi_pack/runners.py", line 145, in _run
    self._symlinks()
  File "/home/acomajuncosa/Documents/ersilia/ersilia/hub/fetch/pack/fastapi_pack/__init__.py", line 26, in _symlinks
    self._dest_bundle_symlink()
  File "/home/acomajuncosa/Documents/ersilia/ersilia/hub/fetch/pack/fastapi_pack/__init__.py", line 20, in _dest_bundle_symlink
    src = os.path.join(bundle_dir, "model")
  File "/home/acomajuncosa/miniconda3/envs/ersilia/lib/python3.10/posixpath.py", line 76, in join
    a = os.fspath(a)
TypeError: expected str, bytes or os.PathLike object, not NoneType

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/acomajuncosa/miniconda3/envs/ersilia/lib/python3.10/site-packages/click/core.py", line 1082, in main
    rv = self.invoke(ctx)
  File "/home/acomajuncosa/miniconda3/envs/ersilia/lib/python3.10/site-packages/click/core.py", line 1697, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/acomajuncosa/miniconda3/envs/ersilia/lib/python3.10/site-packages/click/core.py", line 1443, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/acomajuncosa/miniconda3/envs/ersilia/lib/python3.10/site-packages/click/core.py", line 788, in invoke
    return __callback(*args, **kwargs)
  File "/home/acomajuncosa/Documents/ersilia/ersilia/cli/commands/__init__.py", line 31, in wrapper
    return func(*args, **kwargs)
  File "/home/acomajuncosa/Documents/ersilia/ersilia/cli/commands/fetch.py", line 142, in fetch
    fetch_result = _fetch(mf, model_id)
  File "/home/acomajuncosa/Documents/ersilia/ersilia/cli/commands/fetch.py", line 37, in _fetch
    res = asyncio.run(mf.fetch(model_id))
  File "/home/acomajuncosa/miniconda3/envs/ersilia/lib/python3.10/site-packages/nest_asyncio.py", line 30, in run
    return loop.run_until_complete(task)
  File "/home/acomajuncosa/miniconda3/envs/ersilia/lib/python3.10/site-packages/nest_asyncio.py", line 92, in run_until_complete
    self._run_once()
  File "/home/acomajuncosa/miniconda3/envs/ersilia/lib/python3.10/site-packages/nest_asyncio.py", line 133, in _run_once
    handle._run()
  File "/home/acomajuncosa/miniconda3/envs/ersilia/lib/python3.10/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/home/acomajuncosa/miniconda3/envs/ersilia/lib/python3.10/asyncio/tasks.py", line 232, in __step
    result = coro.send(None)
  File "/home/acomajuncosa/Documents/ersilia/ersilia/hub/fetch/fetch.py", line 366, in fetch
    fr = await self._fetch(model_id)
  File "/home/acomajuncosa/Documents/ersilia/ersilia/hub/fetch/fetch.py", line 333, in _fetch
    self._fetch_not_from_dockerhub(model_id=model_id)
  File "/home/acomajuncosa/Documents/ersilia/ersilia/utils/exceptions_utils/throw_ersilia_exception.py", line 25, in inner_function
    return func(*args, **kwargs)
  File "/home/acomajuncosa/Documents/ersilia/ersilia/hub/fetch/fetch.py", line 225, in _fetch_not_from_dockerhub
    self._fetch_from_fastapi()
  File "/home/acomajuncosa/Documents/ersilia/ersilia/utils/exceptions_utils/throw_ersilia_exception.py", line 25, in inner_function
    return func(*args, **kwargs)
  File "/home/acomajuncosa/Documents/ersilia/ersilia/hub/fetch/fetch.py", line 173, in _fetch_from_fastapi
    mf.fetch(model_id=self.model_id)
  File "/home/acomajuncosa/Documents/ersilia/ersilia/hub/fetch/fetch_fastapi.py", line 185, in fetch
    self._fetch(model_id=model_id)
  File "/home/acomajuncosa/Documents/ersilia/ersilia/hub/fetch/fetch_fastapi.py", line 130, in _fetch
    self._pack()
  File "/home/acomajuncosa/Documents/ersilia/ersilia/hub/fetch/fetch_fastapi.py", line 98, in _pack
    mp.pack()
  File "/home/acomajuncosa/Documents/ersilia/ersilia/hub/fetch/actions/pack_fastapi.py", line 74, in pack
    self._run()
  File "/home/acomajuncosa/Documents/ersilia/ersilia/hub/fetch/actions/pack_fastapi.py", line 56, in _run
    runner.run()
  File "/home/acomajuncosa/Documents/ersilia/ersilia/utils/exceptions_utils/throw_ersilia_exception.py", line 41, in inner_function
    sys.exit(DEFAULT_ERSILIA_ERROR_EXIT_CODE)
SystemExit: 1
`
```

GemmaTuron commented:
indeed Input shape is a legacy field. @arnaucoma24 in the future please add the logs as attchement files or the issues become illegible, thx

GemmaTuron commented:
The input shape field is deprecated and will be eliminated from the new models. 
This problem has been solved in this PR: https://github.com/ersilia-os/ersilia/pull/1621


But the input shape is somewhat still giving problems. @miquelduranfrigola you may need to have a look as I do not know much about how this part of ersilia works. The problem is in this function:
```
    def _get_from_model(self, model_id):
        input_type = self._read_input_from_card(model_id)
        if input_type is None:
            input_type = "naive"
        input_shape = self.shape(model_id)
        self.logger.debug("Input type is: {0}".format(input_type))
        self.logger.debug("Input shape is: {0}".format(input_shape.name))
        module = ".types.{0}".format(input_type)
        self.logger.debug("Importing module: {0}".format(module))
        return importlib.import_module(module, package="ersilia.io").IO(
            input_shape=input_shape
        )

  File "/home/gturon/github/ersilia-os/ersilia/ersilia/io/input.py", line 104, in _get_from_model
    return importlib.import_module(module, package="ersilia.io").IO(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: IO.__init__() got an unexpected keyword argument 'input_shape'
```

GemmaTuron commented:
Okay, more on this error. It apparently enters in this IO class which indeed does not have a input_shape as argument because it is the one located in the `naive.py` file:
```
def _get_from_model(self, model_id):
        input_type = self._read_input_from_card(model_id)
        print("HERE",input_type)
        if input_type is None:
            input_type = "naive"
        input_shape = self.shape(model_id)
        self.logger.debug("Input type is: {0}".format(input_type))
        self.logger.debug("Input shape is: {0}".format(input_shape.name))
        module = ".types.{0}".format(input_type)
        self.logger.debug("Importing module: {0}".format(module))
        return importlib.import_module(module, package="ersilia.io").IO(
            input_shape=input_shape
        )
```

For some reason, the model card is not read (but it is there, in `card.json`), the input_type is None and therefore the IO class is called from the naive file, which only contains:

```
class IO(object):
    def __init__(self):
        pass

    def parse(self, text):
        data = {"key": text, "input": text, "text": text}
        return data
```

If we go back a bit, we see the model is stuck here:

```
15:16:39 | DEBUG    | No need to use Conda!
🚀 Serving model eos6758: mole-representations

   URL: http://0.0.0.0:46401
   PID: 140271
   SRV: conda
   Session: /home/gturon/eos/sessions/session_139865

👉 Run model:
   - run

💁 Information:
   - info

💾 Local cache:
   - Disabled

📈 Tracking:
   - Disabled
```

This is the last printed output. Seemingly of the list of commands:
`ersilia serve eos6758 --no-cache && ersilia example -n 3 -f /home/gturon/eos/dest/eos6758/example_standard_input.csv --simple && ersilia -v run -i /home/gturon/eos/dest/eos6758/example_standard_input.csv -o /home/gturon/eos/dest/eos6758/example_standard_output.csv > /home/gturon/eos/dest/eos6758/standard_run.log 2>&1 && ersilia close`

it is able to serve the model but the example file is not generated and therefore the run does not go an it silently fails, only to crash when the standard_run.log is tried to be read.

I think the source of the problem is that the card is not being read.

GemmaTuron commented:
THE PROBLEM:

Input is not a List in this new model:

 "metadata": {
        "Identifier": "eos6758",
        "Slug": "mole-representations",
        "Status": "In progress",
        "Title": "MolE molecular representation through redundancy reduced embeddings",
        "Description": "MolE representations (Molecular representation through redundancy reduced Embeddings) are task-independent, learned molecular embeddings generated through a self-supervised deep learning approach. They are designed to encode chemically meaningful information about molecules without needing labeled training data.",
        "Source": "Local",
        "Source Type": "External",
        "Task": "Representation",
        "Subtask": "Featurization",
        "Input": "Compound",
        "Input Dimension": 1,
        "Output": "Descriptor",
        "Output Dimension": 1000,
        "Output Consistency": "Fixed",
        "Interpretation": "Vector representation of a molecule",
        "Tag": [
            "Descriptor"
        ],
....
It needs to be parsed as a LIST

GemmaTuron commented:
I don't know why the PR is failing but locally now the model works, I've changed the input and output to lists and solved this small bug: https://github.com/ersilia-os/ersilia/issues/1623

But the output is all None, so something is not working, but this might be due to package installs. Are you 100% sure the dependencies are well listed?

miquelduranfrigola commented:
Thanks both. Any action you need from me now?
